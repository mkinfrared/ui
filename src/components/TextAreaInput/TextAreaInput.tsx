import { TextAreaInputProps } from "./TextAreaInput.type";

const TextAreaInput = ({
  className,
  textAreaRef,
  value,
  ...rest
}: TextAreaInputProps) => {
  const testid = "TextAreaInput";

  if (typeof value !== "undefined") {
    return (
      <textarea
        {...rest}
        className={className}
        data-testid={testid}
        ref={textAreaRef}
        value={value}
      />
    );
  }

  return (
    <textarea
      {...rest}
      className={className}
      data-testid={testid}
      ref={textAreaRef}
    />
  );
};

export default TextAreaInput;
