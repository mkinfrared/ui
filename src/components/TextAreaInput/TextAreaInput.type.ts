import { Ref, TextareaHTMLAttributes } from "react";

type TextAreaAttributes = TextareaHTMLAttributes<HTMLTextAreaElement>;

export type TextAreaInputProps = TextAreaAttributes & {
  textAreaRef?: Ref<HTMLTextAreaElement>;
};
