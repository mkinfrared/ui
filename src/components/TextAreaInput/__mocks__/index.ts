export { default } from "./TextAreaInput.mock";
