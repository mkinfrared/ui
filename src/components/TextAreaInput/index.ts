export { default } from "./TextAreaInput";

export * from "./TextAreaInput.type";
