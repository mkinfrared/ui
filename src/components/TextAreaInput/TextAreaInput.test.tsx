import { render } from "@testing-library/react";

import TextAreaInput from "./TextAreaInput";

describe("<TextAreaInput />", () => {
  const Component = <TextAreaInput />;

  it("should be defined", () => {
    expect(TextAreaInput).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("TextAreaInput");

    expect(element).toBeDefined();
  });

  it("should set value on input if it's not undefined", () => {
    const { getByTestId } = render(<TextAreaInput value="marklar" />);
    const element = getByTestId("TextAreaInput");

    expect(element).toHaveValue("marklar");
  });
});
