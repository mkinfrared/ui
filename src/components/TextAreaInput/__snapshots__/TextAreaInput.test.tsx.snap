// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextAreaInput /> should match the snapshot 1`] = `
<div>
  <textarea
    data-testid="TextAreaInput"
  />
</div>
`;
